### ISSUE TITLE
Suggest the usage of NonZeroU#::from where appropriate

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#7291]
I think Clippy should start suggesting the usage of `NonZeroU#::from` from smaller `NonZeroU#`, as in code like:

```rust
#![allow(unused_variables)]
use std::num::{NonZeroU32, NonZeroU64};

fn main() {
    let x: u64 = 100;
    let y = NonZeroU32::new(10).unwrap();

    // Given code:
    let r1 = x / u64::from(y.get());
    let r2 = x % u64::from(y.get());

    // Clippy should suggest:
    let r1 = x / NonZeroU64::from(y);
    let r2 = x % NonZeroU64::from(y);
}
```

**Comment by y21** [comment]
Closing as this was implemented in #13167 
```
warning: consider using `NonZeroU64::from()` for more efficient and type-safe conversion
  --> y.rs:10:18
   |
10 |     let r1 = x / u64::from(y.get());
   |                  ^^^^^^^^^^^^^^^^^^ help: replace with: `NonZeroU64::from(y)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#non_zero_suggestions
note: the lint level is defined here
  --> y.rs:2:9
   |
2  | #![warn(clippy::non_zero_suggestions)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by leonardo-m** [comment]
Now we have the generic num::NonZero too, so perhaps suggesting this is better?

```rust
#![allow(unused_variables)]
#![warn(clippy::all)]
#![warn(clippy::nursery)]
#![warn(clippy::pedantic)]
use std::num::NonZero;

fn main() {
    let x: u64 = 100;
    let y = const { NonZero::<u32>::new(10).unwrap() };

    // Given code:
    let r1 = x / u64::from(y.get());
    let r2 = x % u64::from(y.get());

    // Clippy could suggest:
    let r1 = x / NonZero::<u64>::from(y);
    let r2 = x % NonZero::<u64>::from(y);
}
```
